src/mscorlib/src/System/UnSafeCharBuffer.cs

```diff
@@ -29,25 +29,43 @@ unsafe internal struct UnSafeCharBuffer{
             m_totalSize = bufferSize;    
             m_length = 0;
         }
-    
+
+        [System.Security.SecuritySafeCritical]  // auto-generated
+        public void AppendChar( char charToAppend ) {
+            if( charToAppend == '\0' ) {
```

Comment:
Question, why did you add this? `\0` is a perfectly valid character in .NET strings, they are length-prefixed as well as null-terminated.